eclair-node/src/main/scala/fr/acinq/eclair/api/handlers/PathFinding.scala

```diff
@@ -19,36 +19,47 @@ package fr.acinq.eclair.api.handlers
 import akka.http.scaladsl.server.{MalformedFormFieldRejection, Route}
 import fr.acinq.bitcoin.Crypto.PublicKey
 import fr.acinq.eclair.api.Service
-import fr.acinq.eclair.api.directives.EclairDirectives
+import fr.acinq.eclair.api.directives.{EclairDirectives, RouteFormat}
 import fr.acinq.eclair.api.serde.FormParamExtractors._
 import fr.acinq.eclair.payment.PaymentRequest
 
+import scala.concurrent.ExecutionContext
+
 trait PathFinding {
   this: Service with EclairDirectives =>
 
   import fr.acinq.eclair.api.serde.JsonSupport.{formats, marshaller, serialization}
 
   val findRoute: Route = postRequest("findroute") { implicit t =>
-    formFields(invoiceFormParam, amountMsatFormParam.?, "pathFindingExperimentName".?) {
-      case (invoice@PaymentRequest(_, Some(amount), _, nodeId, _, _), None, pathFindingExperimentName_opt) =>
-        complete(eclairApi.findRoute(nodeId, amount, pathFindingExperimentName_opt, invoice.routingInfo))
-      case (invoice, Some(overrideAmount), pathFindingExperimentName_opt) =>
-        complete(eclairApi.findRoute(invoice.nodeId, overrideAmount, pathFindingExperimentName_opt, invoice.routingInfo))
+
+    implicit val ec: ExecutionContext = actorSystem.dispatcher
```

Comment:
Why is that needed? Is it because of the `map` call?
Can we lift this outside of each `val` directly in the `trait` or is it impossible?